### PR TITLE
dash/datagrid2-td-resizing-scrolling

### PR DESCRIPTION
**Pull request by sebastianbochan**
- [x] - data generator (S)
- [x] - fix init width bug in the `<td>` and `<th>` - add ellipsis (S)
- [x] - handlers rendering (S)
- [x] - basic resizer logic / events (S)
- [x] - Introduction of options for managing column width distribution in the table.
- [x] - Implementing resize logic for `full` distribution.
- [x] - Moving the code to the `ColumnsResizer` class.
- [x] - Implementing resize logic for `fixed` distribution and horizontal scroll.
- [x] - Optimizing the code.

### CODE REVIEW

File: css/datagrid/datagrid2.css [hunk 1]
```diff
@@ -126,6 +133,8 @@
     border-right: 1px solid #00ff00;
     white-space: nowrap;
     padding-left: 4px;
+    text-overflow: ellipsis;
```
Comment (sebastianbochan) [inline css/datagrid/datagrid2.css:136]: why do we need it here?

File: css/datagrid/datagrid2.css [hunk 2]
```diff
@@ -143,3 +152,30 @@
...
+    z-index: 10;
+}
+
+.highcharts-dg-col-resizer::after {
+    content: "";
+    position: absolute;
+    display: block;
+    height: 20px;
+    width: 4px;
+    margin-top: -10px;
```
Comment (sebastianbochan) [inline css/datagrid/datagrid2.css:173]: ?

File: ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    /**
+     * The minimum width of a column.
+     */
+    private static readonly MIN_COLUMN_WIDTH = 20;
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts:50]: do we need it as static?

File: ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+
+    /**
+     * The width of the next column when dragging started.
+     */
+    private nextColumnStartWidth?: number;
+
+    /**
+     * The handles and their mouse down event listeners.
+     */
+    private handles: Array<[HTMLElement, (e: MouseEvent) => void]> = [];
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts:91]: smart ;)

File: ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+
+    /**
+     * The column being dragged.
+     */
+    private draggedColumn?: DataGridColumn;
+
+    /**
+     * The column right next to the dragged column.
+     */
+    private nextColumn?: DataGridColumn;
```
Comment (sebastianbochan) [inline ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts:66]: Do we need it? The next column is `columns[index + 1]`, especially that we have an index of the current one.

File: tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        /* Visit https://aka.ms/tsconfig to read more about this file */
+
+        /* Projects */
+        // "incremental": true,                              /* Save .tsbuildinfo files to allow for incremental compilation of projects. */
+        // "composite": true,                                /* Enable constraints that allow a TypeScript project to be used with project references. */
+        "composite": true,
+        // "tsBuildInfoFile": "./.tsbuildinfo",              /* Specify the path to .tsbuildinfo incremental compilation file. */
+        // "disableSourceOfProjectReferenceRedirect": true,  /* Disable preferring source files instead of declaration files when referencing composite projects. */
+        // "disableSolutionSearching": true,                 /* Opt a project out of multi-project reference checking when editing. */
+        // "disableReferencedProjectLoad": true,             /* Reduce the number of projects loaded automatically by TypeScript. */
```
Comment (sebastianbochan) [inline tsconfig.json:12]: I guess that it's an accidental change of the whole `tsconfig.json`

File: ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts (hunk 3 above)
Comment (vazonik) [reply ts/DataGrid/DataGrid2/Actions/ColumnsResizer.ts:50]: This is a global setting and I think static fields fit better for this. There is no need to copy this value for each instance of the class.

File: css/datagrid/datagrid2.css (hunk 1 above)
Comment (vazonik) [reply css/datagrid/datagrid2.css:136]: Just a matter of styling, will probably be changed during the final styling of the DataGrid.

File: css/datagrid/datagrid2.css (hunk 2 above)
Comment (vazonik) [reply css/datagrid/datagrid2.css:173]: Like above, will probably be changed during the final styling of the DataGrid.

**Pull request merged**